docs/tools/tools_intro.md

```diff
@@ -44,3 +45,38 @@ For more information, please see the [Online Compiler page](developing-mbed-onli
 
 You can export your project from any of our tools to third party tools. For instructions, as well as tool-specific information, see [the Exporting to third party toolchains page](exporting.html).
 
+#### Forcing compilation with ARM Compiler 5 for targets already supporting ARM Compiler 6
```

Comment:
Can we make this a note and shorten it?

Maybe something like:

<span class="note"> **Note:** We encourage you to switch to Arm Compiler 6 soon because we will deprecate Arm Compiler 5 support in the future. However, if you need to update to Mbed OS 5.12 but still require compiling with Arm Compiler 5 until you are in possession of Arm Compiler 6, we provide methods to override the Arm toolchain version. If you do this, your target may not be able to compile with Arm Compiler 5, or you may see undefined behaviors.</span>

To force Arm Compiler 5, you can use the following options:

- Create or update your `mbed_app.json` with:

```
{
  "target_overrides": {
      "*": {
          "target.supported_toolchains": ["ARMC5", "GCC_ARM", "IAR"]
      }
  }
}
```

- Modify the `supported_toolchains` entry in targets.json to replace all `ARM`, `ARMC6` entries with `ARMC5`: 

```
"MY_TARGET_NAME": {
        "supported_form_factors": [...],
        "core": "Cortex-M4",
        "supported_toolchains": ["ARMC5", "GCC_ARM", "IAR"],
        ...
}
```

Reply:
See my comment below about `targets.json` modifications. I think that bit may need to move to target porting.

Reply:
@ChiefBureaucraticOfficer @screamerbg - Please review if the shortened note of the content provided by @AnotherButler above works. 

@theotherjimmy - I don't think this belongs in porting section just because we refer targets.json. Also it makes hard to find if its in porting section as this would be the page most people would land on to find compiler info.

Reply:
Should a modification above to mbed_app.json be enough? Just testing mbed-os-blinky with that on mbed OS master (b80c961da) and it still fails to build on ARMC5.
```
mbed compile -m K64F -t ARMC5
<..>
make.py: error: argument -t/--tool: ARMC5 is not a supported toolchain. Supported toolchains are:
ARM,     ARMC6,   GCC_ARM, IAR,     uARM     
[mbed] ERROR: "/usr/bin/python" returned error.
```

Reply:
@TeroJaasko - Yes, please follow the changes as mentioned here to enable ARMC5 and try. By the way "-t ARMC5" is an invalid option. After adding ARMC5 to supported_toolchains you should still do "-t ARM". Please let me know if you need help with making the changes.

Reply:
I already copied the ```"target.supported_toolchains": ["ARMC5", "GCC_ARM", "IAR"]``` to mbed_app.json and the results are no better with "-t ARM".
```
mbed compile -m K64F -t ARM
<..>
make.py: error: Could not find executable for ARMC6.
```

Diff on mbed-os-example-blinky:
```
mbed-os-example-blinky$ git diff
diff --git i/mbed_app.json w/mbed_app.json
index 3b191ff..c106c06 100644
--- i/mbed_app.json
+++ w/mbed_app.json
@@ -1,6 +1,7 @@
 {
     "target_overrides": {
         "*": {
+            "target.supported_toolchains": ["ARMC5", "GCC_ARM", "IAR"],
             "platform.stack-stats-enabled": true,
             "platform.heap-stats-enabled": true,
             "platform.cpu-stats-enabled": true,
```

Reply:
@TeroJaasko - Although we tested this in past, I tried the same thing again with the exact version you are using(b80c961da) and it appears to be working fine on my machine, and yes, the compilation works with ARMC5. And yet, I don't see anything wrong with what you are trying to do.

Reply:
I also tried the same on Ubuntu and that worked fine as well. So I wonder whats causing your failure :-(

Reply:
Have you tried removing ARMC6 from paths? I don't have ARMC6 at my paths by default, there are at least one ticket open where wrong compiler is used: https://github.com/ARMmbed/mbed-os/issues/10069

Reply:
Ah ok, that makes sense if you dont have ARMC6 configured. I believe the tools would check for ARMC6 as in this PR https://github.com/ARMmbed/mbed-os/pull/10044 , have you tried adding ARMC6 and see if that fixes the problem?